parser.py

```diff
@@ -9,26 +9,35 @@
 import sys
 from os import listdir
 from os.path import isfile, join
+from collections import defaultdict
+import json
 nltk.download('punkt')
 
-def GetLanguageId(language):
-    language_id_list = logic.SQLQuery("select Lang_ID from lang_ref where Lang_Desc = '"+ language +"'")
-
-    # Condition for language not present in database
-    if len(language_id_list) == 0:
-        return -1
-    else:
-        return language_id_list[0]
+def CreateIndexing(sentence, line_id, dictionary):
+    word_list = sentence.split()
+    # number of parts formed for each tagged word in the corpus(word/pos)
+    tagged_word_parts = 2   
+    for word in word_list:
+        # skip untagged words
+        if(len(word.split('/')) != tagged_word_parts):
+            continue
+        subtag = (word.split('/')[1]).upper()
+        # ignore -TL suffix in subtags since they are just to indicate that the word occurs in title
+        subtag = subtag.split('-TL')[0]
+        if(subtag in logic.english_pos_mapping):
+            generalized_pos = logic.english_pos_mapping[subtag]
```

Comment:
Does this return the generalized tag, or the subtag?

Reply:
It returns the generalized tag